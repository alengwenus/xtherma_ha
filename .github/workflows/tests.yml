name: Tests

on:
  push:
    branches:
      - main
  pull_request:
  schedule:
    - cron: "0 0 * * *"

env:
  DEFAULT_PYTHON: "3.13"

jobs:
  pre-commit:
    runs-on: "ubuntu-latest"
    name: Pre-commit and pytest
    steps:
      - name: Check out the repository
        uses: actions/checkout@v5

      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
        uses: actions/setup-python@v6
        with:
          python-version: ${{ env.DEFAULT_PYTHON }}
          cache: pip

      - name: Install Python requirements
        run: |
          pip install -r requirements.txt -r dev-requirements.txt

      - name: Run pre-commit on all files
        run: |
          pre-commit run --all-files --show-diff-on-failure --color=always

      - name: Run pytest
        run: |
          python -m pytest --timeout=10 tests

  hacs:
    runs-on: "ubuntu-latest"
    name: HACS
    if: github.repository == 'Xtherma/xtherma_ha'
    steps:
      - name: Check out the repository
        uses: "actions/checkout@v5"

      - name: HACS validation
        uses: "hacs/action@22.5.0"
        with:
          category: "integration"
          ignore: "topics"

  hassfest:
    runs-on: "ubuntu-latest"
    name: Hassfest
    steps:
      - name: Check out the repository
        uses: "actions/checkout@v5"

      - name: Hassfest validation
        uses: "home-assistant/actions/hassfest@master"
