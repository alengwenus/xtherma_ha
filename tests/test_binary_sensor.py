"""Tests for the Xtherma sensor platform."""

from unittest.mock import patch

import pytest
from homeassistant.const import Platform
from pytest_homeassistant_custom_component.common import snapshot_platform

from tests.conftest import init_integration, init_modbus_integration
from tests.helpers import (
    get_binary_sensor_platform,
    provide_modbus_data,
    provide_rest_data,
)

BINARY_SENSOR_ENTITY_ID_PK = (
    "binary_sensor.test_entry_xtherma_config_pk_circulation_pump_enabled"
)
BINARY_SENSOR_ENTITY_ID_PWW = (
    "binary_sensor.test_entry_xtherma_config_pww_circulation_pump_hot_water_enabled"
)


@pytest.mark.parametrize("mock_rest_api_client", provide_rest_data(), indirect=True)
async def test_setup_binary_sensor_rest_api(
    hass, entity_registry, snapshot, mock_rest_api_client
) -> None:
    """Test the setup of binary sensor platform using REST API."""
    with patch("custom_components.xtherma_fp._PLATFORMS", [Platform.BINARY_SENSOR]):
        entry = await init_integration(hass, mock_rest_api_client)

    await snapshot_platform(hass, entity_registry, snapshot, entry.entry_id)


@pytest.mark.parametrize("mock_modbus_tcp_client", provide_modbus_data(), indirect=True)
async def test_setup_binary_sensor_modbus_tcp(
    hass, entity_registry, snapshot, mock_modbus_tcp_client
) -> None:
    """Test the setup of binary sensor platform using MODBUS TCP."""
    with patch("custom_components.xtherma_fp._PLATFORMS", [Platform.BINARY_SENSOR]):
        entry = await init_modbus_integration(hass, mock_modbus_tcp_client)

    await snapshot_platform(hass, entity_registry, snapshot, entry.entry_id)


@pytest.mark.parametrize("mock_rest_api_client", provide_rest_data(), indirect=True)
async def test_binary_sensor_state(hass, mock_rest_api_client):
    await init_integration(hass, mock_rest_api_client)
    pk = hass.states.get(BINARY_SENSOR_ENTITY_ID_PK)
    assert pk.state == "off"
    pww = hass.states.get(BINARY_SENSOR_ENTITY_ID_PWW)
    assert pww.state == "on"


@pytest.mark.parametrize("mock_rest_api_client", provide_rest_data(), indirect=True)
async def test_binary_sensor_icon(hass, mock_rest_api_client):
    await init_integration(hass, mock_rest_api_client)
    platform = get_binary_sensor_platform(hass)
    pk = hass.states.get(BINARY_SENSOR_ENTITY_ID_PK)
    assert pk.state == "off"
    e_pk = platform.entities.get(pk.entity_id)
    assert e_pk is not None
    assert e_pk.icon == "mdi:pump-off"
    pww = hass.states.get(BINARY_SENSOR_ENTITY_ID_PWW)
    assert pww.state == "on"
    e_pww = platform.entities.get(pww.entity_id)
    assert e_pww is not None
    assert e_pww.icon == "mdi:pump"


@pytest.mark.parametrize("mock_rest_api_client", provide_rest_data(), indirect=True)
async def test_binary_sensor_name(hass, mock_rest_api_client):
    """Check if binary sensors have a proper (translated) names."""
    await init_integration(hass, mock_rest_api_client)
    platform = get_binary_sensor_platform(hass)
    state = hass.states.get(BINARY_SENSOR_ENTITY_ID_PWW)
    entity = platform.entities.get(state.entity_id)
    assert entity is not None
    assert entity.name == "[PWW] Circulation pump hot water enabled"
