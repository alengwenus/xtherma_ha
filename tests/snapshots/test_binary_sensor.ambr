# serializer version: 1
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_evu_status-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_evu_status',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:electric-switch-closed',
    'original_name': 'EVU status',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'evu',
    'unique_id': 'test_entry_xtherma-evu',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_evu_status-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config EVU status',
      'icon': 'mdi:electric-switch-closed',
      'parameter': 'evu',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_evu_status',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pk1_circulation_pump_circuit_1_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pk1_circulation_pump_circuit_1_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PK1] Circulation pump circuit 1 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pk1',
    'unique_id': 'test_entry_xtherma-pk1',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pk1_circulation_pump_circuit_1_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_modbus_config [PK1] Circulation pump circuit 1 enabled',
      'icon': 'mdi:pump-off',
      'parameter': 'pk1',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pk1_circulation_pump_circuit_1_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pk2_circulation_pump_circuit_2_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pk2_circulation_pump_circuit_2_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PK2] Circulation pump circuit 2 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pk2',
    'unique_id': 'test_entry_xtherma-pk2',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pk2_circulation_pump_circuit_2_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_modbus_config [PK2] Circulation pump circuit 2 enabled',
      'icon': 'mdi:pump-off',
      'parameter': 'pk2',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pk2_circulation_pump_circuit_2_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pk_circulation_pump_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pk_circulation_pump_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PK] Circulation pump enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pk',
    'unique_id': 'test_entry_xtherma-pk',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pk_circulation_pump_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_modbus_config [PK] Circulation pump enabled',
      'icon': 'mdi:pump-off',
      'parameter': 'pk',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pk_circulation_pump_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pww_circulation_pump_hot_water_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pww_circulation_pump_hot_water_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PWW] Circulation pump hot water enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pww',
    'unique_id': 'test_entry_xtherma-pww',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_pww_circulation_pump_hot_water_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_modbus_config [PWW] Circulation pump hot water enabled',
      'icon': 'mdi:pump',
      'parameter': 'pww',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_pww_circulation_pump_hot_water_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_ss14a_enwg_state-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_ss14a_enwg_state',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'ยง14a EnWG state',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '14a',
    'unique_id': 'test_entry_xtherma-14a',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_ss14a_enwg_state-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config ยง14a EnWG state',
      'parameter': '14a',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_ss14a_enwg_state',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_system_ok-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_system_ok',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:alert',
    'original_name': 'System ok',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'error',
    'unique_id': 'test_entry_xtherma-error',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_modbus_tcp[mock_modbus_tcp_client0][binary_sensor.test_entry_xtherma_modbus_config_system_ok-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_modbus_config System ok',
      'icon': 'mdi:check',
      'parameter': 'error',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_modbus_config_system_ok',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_evu_status-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_evu_status',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:electric-switch-closed',
    'original_name': 'EVU status',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'evu',
    'unique_id': 'test_entry_xtherma-evu',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_evu_status-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config EVU status',
      'icon': 'mdi:electric-switch-closed',
      'parameter': 'evu',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_evu_status',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pk1_circulation_pump_circuit_1_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pk1_circulation_pump_circuit_1_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PK1] Circulation pump circuit 1 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pk1',
    'unique_id': 'test_entry_xtherma-pk1',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pk1_circulation_pump_circuit_1_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_config [PK1] Circulation pump circuit 1 enabled',
      'icon': 'mdi:pump-off',
      'parameter': 'pk1',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pk1_circulation_pump_circuit_1_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pk2_circulation_pump_circuit_2_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pk2_circulation_pump_circuit_2_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PK2] Circulation pump circuit 2 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pk2',
    'unique_id': 'test_entry_xtherma-pk2',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pk2_circulation_pump_circuit_2_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_config [PK2] Circulation pump circuit 2 enabled',
      'icon': 'mdi:pump-off',
      'parameter': 'pk2',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pk2_circulation_pump_circuit_2_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pk_circulation_pump_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pk_circulation_pump_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PK] Circulation pump enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pk',
    'unique_id': 'test_entry_xtherma-pk',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pk_circulation_pump_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_config [PK] Circulation pump enabled',
      'icon': 'mdi:pump-off',
      'parameter': 'pk',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pk_circulation_pump_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pww_circulation_pump_hot_water_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pww_circulation_pump_hot_water_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:pump-off',
    'original_name': '[PWW] Circulation pump hot water enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'pww',
    'unique_id': 'test_entry_xtherma-pww',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_pww_circulation_pump_hot_water_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_config [PWW] Circulation pump hot water enabled',
      'icon': 'mdi:pump',
      'parameter': 'pww',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_pww_circulation_pump_hot_water_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_ss14a_enwg_state-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_ss14a_enwg_state',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'ยง14a EnWG state',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '14a',
    'unique_id': 'test_entry_xtherma-14a',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_ss14a_enwg_state-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config ยง14a EnWG state',
      'parameter': '14a',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_ss14a_enwg_state',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_system_ok-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_system_ok',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.RUNNING: 'running'>,
    'original_icon': 'mdi:alert',
    'original_name': 'System ok',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'error',
    'unique_id': 'test_entry_xtherma-error',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_binary_sensor_rest_api[mock_rest_api_client0][binary_sensor.test_entry_xtherma_config_system_ok-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'running',
      'friendly_name': 'test_entry_xtherma_config System ok',
      'icon': 'mdi:check',
      'parameter': 'error',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.test_entry_xtherma_config_system_ok',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
